doc/develop/test/blackbox.rst

```diff
@@ -0,0 +1,180 @@
+.. _blackbox:
+
+======================
+Twister blackbox tests
+======================
+
+Basics
+======
+
+Twister blackbox tests are written in python, with auxiliary test data following
+whichever format it was in originally. They are wholly contained in the
+`scripts/tests/twister_blackbox/` directory inside zephyr and prepended with `test_`.
+
+Blackbox tests should not be aware of the internal twister code. Instead, they should
```

Comment:
Black Box Testing Techniques usually has below:

<html>
<body>
<!--StartFragment-->
Decision Table Technique | Decision Table Technique is a systematic approach where various input combinations and their respective system behavior are captured in a tabular form. It is appropriate for the functions that have a logical relationship between two and more than two inputs.
-- | --
Boundary Value Technique | Boundary Value Technique is used to test boundary values, boundary values are those that contain the upper and lower limit of a variable. It tests, while entering boundary value whether the software is producing correct output or not.
State Transition Technique | State Transition Technique is used to capture the behavior of the software application when different input values are given to the same function. This applies to those types of applications that provide the specific number of attempts to access the application.
All-pair Testing Technique | All-pair testing Technique is used to test all the possible discrete combinations of values. This combinational method is used for testing the application that uses checkbox input, radio button input, list box, text box, etc.
Cause-Effect Technique | Cause-Effect Technique underlines the relationship between a given result and all the factors affecting the result.It is based on a collection of requirements.
Equivalence Partitioning Technique | Equivalence partitioning is a technique of software testing in which input data divided into partitions of valid and invalid values, and it is mandatory that all partitions must exhibit the same behavior.
Error Guessing Technique | Error guessing is a technique in which there is no specific method for identifying the error. It is based on the experience of the test analyst, where the tester uses the experience to guess the problematic areas of the software.
Use Case Technique | Use case Technique used to identify the test cases from the beginning to the end of the system as per the usage of the system. By using this technique, the test team creates a test scenario that can exercise the entire software based on the functionality of each function from start to end.

<!--EndFragment-->
</body>
</html>[Decision Table Technique](https://www.javatpoint.com/decision-table-technique-in-black-box-testing)	Decision Table Technique is a systematic approach where various input combinations and their respective system behavior are captured in a tabular form. It is appropriate for the functions that have a logical relationship between two and more than two inputs.
[Boundary Value Technique](https://www.javatpoint.com/boundary-value-analysis-in-black-box-testing)	Boundary Value Technique is used to test boundary values, boundary values are those that contain the upper and lower limit of a variable. It tests, while entering boundary value whether the software is producing correct output or not.
[State Transition Technique](https://www.javatpoint.com/state-transition-technique-in-black-box-testing)	State Transition Technique is used to capture the behavior of the software application when different input values are given to the same function. This applies to those types of applications that provide the specific number of attempts to access the application.
[All-pair Testing Technique](https://www.javatpoint.com/all-pairs-testing-technique-in-black-box-testing)	All-pair testing Technique is used to test all the possible discrete combinations of values. This combinational method is used for testing the application that uses checkbox input, radio button input, list box, text box, etc.
[Cause-Effect Technique](https://www.javatpoint.com/cause-and-effect-graph-technique-in-black-box-testing)	Cause-Effect Technique underlines the relationship between a given result and all the factors affecting the result.It is based on a collection of requirements.
[Equivalence Partitioning Technique](https://www.javatpoint.com/equivalence-partitioning-technique-in-black-box-testing)	Equivalence partitioning is a technique of software testing in which input data divided into partitions of valid and invalid values, and it is mandatory that all partitions must exhibit the same behavior.
[Error Guessing Technique](https://www.javatpoint.com/error-guessing-technique-in-black-box-testing)	Error guessing is a technique in which there is no specific method for identifying the error. It is based on the experience of the test analyst, where the tester uses the experience to guess the problematic areas of the software.
[Use Case Technique](https://www.javatpoint.com/use-case-technique-in-black-box-testing)	Use case Technique used to identify the test cases from the beginning to the end of the system as per the usage of the system. By using this technique, the test team creates a test scenario that can exercise the entire software based on the functionality of each function from start to end.

what do we plan to use here?

Reply:
Current approach is not directly based on any blackbox testing framework. Current aim is to cover every Twister flag individually. We do not aim to have a full combinatorial coverage of flag combinations. That would make it most similar to the Cause-Effect Technique from the table. However, applying typical blackbox testing techniques is harder, as practically all Twister flags and setting have scarce descriptions of their desired results - a far cry from a well-defined input-output contracts common to Decision Table Technique.